Apps/WebClient/src/ClientApp/src/models/timeline/bcCancerScreeningTimelineEntry.ts

```diff
@@ -1,35 +1,63 @@
 import { EntryType } from "@/constants/entryType";
 import { DateWrapper, StringISODate } from "@/models/dateWrapper";
-import { BcCancerScreening } from "@/models/patientDataResponse";
+import {
+    BcCancerScreening,
+    BcCancerScreeningType,
+} from "@/models/patientDataResponse";
 import TimelineEntry from "@/models/timeline/timelineEntry";
 import { UserComment } from "@/models/userComment";
 export default class BcCancerScreeningTimelineEntry extends TimelineEntry {
-    public title: string;
-    public documentType: string;
+    public title!: string;
+    public documentType!: string;
     public fileId: string;
-    public resultDate: StringISODate;
-    public programName: string;
+    public entryDate!: StringISODate;
+    public subtitle: string;
+    public callToActionText!: string;
+    public screeningType: BcCancerScreeningType;
+    public isResult!: boolean;
+    public fileName!: string;
+    public eventText!: string;
 
     private getComments: (entryId: string) => UserComment[] | null;
 
     public constructor(
         model: BcCancerScreening,
         getComments: (entryId: string) => UserComment[] | null
     ) {
+        const isResult = model.eventType === BcCancerScreeningType.Result;
+        const date = isResult ? model.resultDateTime : model.eventDateTime;
         super(
-            model.id ?? `cancerScreening-${model.resultDateTime}`,
+            model.id ?? `cancerScreening-${date}`,
             EntryType.BcCancerScreening,
-            new DateWrapper(model.resultDateTime, { isUtc: true })
+            new DateWrapper(date, { isUtc: true })
         );
-
-        this.title = "BC Cancer Result";
-        this.documentType = "Screening results";
-        this.programName = model.programName;
+        this.isResult = isResult;
+        this.entryDate = date;
+        this.subtitle = `Programe: ${model.programName}`;
         this.fileId = model.fileId;
-        this.resultDate = model.resultDateTime;
+        this.screeningType = model.eventType;
+        this.fileName = this.isResult
+            ? "bc_cancer_result"
+            : "bc_cancer_screening";
+        this.eventText = this.isResult
+            ? "BC Cancer Result PDF"
+            : "BC Cancer Screening PDF";
```

Comment:
These assignments could be moved into the `setEntryProperties()` function so they're grouped with the other properties that change based on event type.